File: stage-common/src/main/java/io/github/solusmods/eternalcore/stage/EternalCoreStage.java
package io.github.solusmods.eternalcore.stage;

import io.github.solusmods.eternalcore.stage.impl.StageStorage;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import net.minecraft.resources.ResourceLocation;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static io.github.solusmods.eternalcore.stage.ModuleConstants.MOD_ID;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class EternalCoreStage {
    public static final Logger LOG = LoggerFactory.getLogger("EternalCore - Stage");

    public static ResourceLocation create(String path) {
        return ResourceLocation.fromNamespaceAndPath(MOD_ID, path);
    }

    public static void init(){
        StageStorage.init();
    }

}

File: stage-common/src/main/kotlin/io/github/solusmods/eternalcore/stage/EternalCoreStage.kt
package io.github.solusmods.eternalcore.stage

import io.github.solusmods.eternalcore.stage.impl.StageRegistry
import io.github.solusmods.eternalcore.stage.impl.StageStorage
import io.github.solusmods.eternalcore.stage.impl.network.StagesNetwork
import net.minecraft.resources.ResourceLocation
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object EternalCoreStage {
    val LOG: Logger? = LoggerFactory.getLogger("EternalCore - Stage")

    fun create(path: String): ResourceLocation {
        return ResourceLocation.fromNamespaceAndPath(ModuleConstants.MOD_ID, path)
    }

    @JvmStatic
    fun init() {
        StageStorage.Companion.init()
        StagesNetwork.init()
        StageRegistry.init()
    }
}
